tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/error_match_pattern_negative.bal

```diff
@@ -41,7 +41,7 @@ function testNegative2() {
         error("Message") | error("Message") => {} // unreachable pattern
         error(_) | error(_) => {} // unreachable pattern
         error(var a) | error(var b) => {} // unreachable pattern
-        error("Message", error("Msg")) | error("Message", error("Msg")) => {} // unreachable pattern
+        error("Message", error("foo")) | error("Message", error("foo")) => {} // unreachable pattern
```

Comment:
The purpose of the line is to check whether 2nd pattern is unreachable or not. 
Because this PR fixes some issues in match-stmt, without this change following lines fails with `unreachable pattern`. 
To run the following lines as expected I did this change.

Reply:
How does changing "Msg" to "foo" fix this? It's just a literal? And the behaviour should be the same for both?

Reply:
If we did not change it, L:45 and L:46 will have `unreachable pattern` error

Reply:
No, it is available in https://github.com/ballerina-platform/ballerina-lang/blob/ca7041f32e9b3d204319856babade1aac4c63128/tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/matchstmt/MatchStmtErrorMatchPatternTest.java#L158